core/src/services/yandex_disk/backend.rs

```diff
@@ -0,0 +1,328 @@
+        } else {
+            Some(root.to_string())
+        };
+
+        self
+    }
+
+    /// yandex disk oauth access_token.
+    /// visit https://yandex.com/dev/id/doc/en/ to get access_token
+    ///
+    /// It is required. e.g. `xxxx`
+    pub fn access_token(&mut self, access_token: &str) -> &mut Self {
+        self.config.access_token = access_token.to_string();
+
+        self
+    }
+
+    /// Specify the http client that used by this service.
+    ///
+    /// # Notes
+    ///
+    /// This API is part of OpenDAL's Raw API. `HttpClient` could be changed
+    /// during minor updates.
+    pub fn http_client(&mut self, client: HttpClient) -> &mut Self {
+        self.http_client = Some(client);
+        self
+    }
+}
+
+impl Builder for YandexDiskBuilder {
+    const SCHEME: Scheme = Scheme::YandexDisk;
+    type Accessor = YandexDiskBackend;
+
+    /// Converts a HashMap into an YandexDiskBuilder instance.
+    ///
+    /// # Arguments
+    ///
+    /// * `map` - A HashMap containing the configuration values.
+    ///
+    /// # Returns
+    ///
+    /// Returns an instance of YandexDiskBuilder.
+    fn from_map(map: HashMap<String, String>) -> Self {
+        // Deserialize the configuration from the HashMap.
+        let config = YandexDiskConfig::deserialize(ConfigDeserializer::new(map))
+            .expect("config deserialize must succeed");
+
+        // Create an YandexDiskBuilder instance with the deserialized config.
+        YandexDiskBuilder {
+            config,
+            http_client: None,
+        }
+    }
+
+    /// Builds the backend and returns the result of YandexDiskBackend.
+    fn build(&mut self) -> Result<Self::Accessor> {
+        debug!("backend build started: {:?}", &self);
+
+        let root = normalize_root(&self.config.root.clone().unwrap_or_default());
+        debug!("backend use root {}", &root);
+
+        // Handle oauth access_token.
+        if self.config.access_token.is_empty() {
+            return Err(
+                Error::new(ErrorKind::ConfigInvalid, "access_token is empty")
+                    .with_operation("Builder::build")
+                    .with_context("service", Scheme::YandexDisk),
+            );
+        }
+
+        let client = if let Some(client) = self.http_client.take() {
+            client
+        } else {
+            HttpClient::new().map_err(|err| {
+                err.with_operation("Builder::build")
+                    .with_context("service", Scheme::YandexDisk)
+            })?
+        };
+
+        Ok(YandexDiskBackend {
+            core: Arc::new(YandexDiskCore {
+                root,
+                access_token: self.config.access_token.clone(),
+                client,
+            }),
+        })
+    }
+}
+
+/// Backend for YandexDisk services.
+#[derive(Debug, Clone)]
+pub struct YandexDiskBackend {
+    core: Arc<YandexDiskCore>,
+}
+
+#[async_trait]
+impl Accessor for YandexDiskBackend {
+    type Reader = IncomingAsyncBody;
```

Comment:
Please ensure that the order of those types and functions matches that of `Accessor`.

Reply:
fixed